crud/src/test/java/com/redhat/lightblue/assoc/ep/ExecutionPlanTest.java

```diff
@@ -0,0 +1,260 @@
+   //        Once bound, this should be rewritten as:
+   //           {field:localized(rf),op:=,rvalues:x.i.y}
+   //        In other words, array elem match with array X and field Y is treated like a search on X.*.Y
+
+   //       */
+   //      Projection p=projection("[{'field':'*','recursive':1},{'field':'us','recursive':1}]");
+   //      QueryExpression q=query("{'array':'us.*.authentications','elemMatch':{ '$and':[ { 'field':'principal','op':'$in','values':['a']}, {'field':'providerName','op':'$eq','rvalue':'p'} ] } }");
+   //      GMD gmd=new GMD(p,null);
+   //      CompositeMetadata md=CompositeMetadata.buildCompositeMetadata(getMd("composite/L.json"),gmd);
+        
+   //      Set<CompositeMetadata> minimalTree=new HashSet<>();
+   //      minimalTree.add(md);
+   //      minimalTree.add(md.getChildMetadata(new Path("us")));
+   //      QueryPlan searchQP=new QueryPlanChooser(md,
+   //                                              new BruteForceQueryPlanIterator(),
+   //                                              new IndexedFieldScorer(),
+   //                                              q,
+   //                                              minimalTree).choose();
+   //      QueryPlan retrievalQP=new QueryPlanChooser(md,
+   //                                                 new First(),
+   //                                                 new SimpleScorer(),
+   //                                                 q,
+   //                                                 minimalTree).choose();
+        
+   //      ExecutionPlan ep=new ExecutionPlan(p,null,null,null,md,searchQP,retrievalQP);
+   //      ObjectNode j=(ObjectNode)ep.toJson();
+   //      ObjectNode root=(ObjectNode)j.get("root");
+        
+   //      System.out.println("rev_search_with_arraycond");
+   //      System.out.println(ep.toString());
+   //      System.out.println("minimal tree:"+minimalTree);
+
+   //      ArrayNode children=(ArrayNode)root.get("children");
+   //      Assert.assertEquals(1,children.size());
+   //      ObjectNode u1=(ObjectNode)children.get(0);
+   //      Assert.assertEquals("U_1",u1.get("name").asText());
+   //      children=(ArrayNode)u1.get("children");
+   //      Assert.assertEquals(1,children.size());
+   //      ObjectNode l0=(ObjectNode)children.get(0);
+   //      Assert.assertEquals("L_0",l0.get("name").asText());
+   //      children=(ArrayNode)l0.get("children");
+   //      Assert.assertEquals(1,children.size());
+   //      ObjectNode f=(ObjectNode)children.get(0);
+   //      children=(ArrayNode)f.get("children");
+   //      Assert.assertEquals(1,children.size());
+   //      l0=(ObjectNode)children.get(0);
+   //      Assert.assertEquals("L_0",l0.get("name").asText());
+   //      children=(ArrayNode)l0.get("children");
+   //      Assert.assertEquals(1,children.size());
+   //      u1=(ObjectNode)children.get(0);
+   //      Assert.assertEquals("U_1",u1.get("name").asText());
```

Comment:
That test was written for a previous incarnation of ExecutionPlan. I have
to rewrite the checks to make it work with the latest version.

On Wed, Mar 30, 2016 at 11:28 AM, Brandon Vulaj <notifications@github.com>
wrote:

> In crud/src/test/java/com/redhat/lightblue/assoc/ep/ExecutionPlanTest.java
> <https://github.com/lightblue-platform/lightblue-core/pull/608#discussion_r57928934>
> :
>
> > +   //      Assert.assertEquals("U_1",u1.get("name").asText());
> > +   //      children=(ArrayNode)u1.get("children");
> > +   //      Assert.assertEquals(1,children.size());
> > +   //      ObjectNode l0=(ObjectNode)children.get(0);
> > +   //      Assert.assertEquals("L_0",l0.get("name").asText());
> > +   //      children=(ArrayNode)l0.get("children");
> > +   //      Assert.assertEquals(1,children.size());
> > +   //      ObjectNode f=(ObjectNode)children.get(0);
> > +   //      children=(ArrayNode)f.get("children");
> > +   //      Assert.assertEquals(1,children.size());
> > +   //      l0=(ObjectNode)children.get(0);
> > +   //      Assert.assertEquals("L_0",l0.get("name").asText());
> > +   //      children=(ArrayNode)l0.get("children");
> > +   //      Assert.assertEquals(1,children.size());
> > +   //      u1=(ObjectNode)children.get(0);
> > +   //      Assert.assertEquals("U_1",u1.get("name").asText());
>
> Can we just @Ignore <https://github.com/Ignore> instead of all the
> comments?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-core/pull/608/files/4ed1ac30c567cdd5b0c9b29591a5eac893d9a412#r57928934>
>